components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQConsumer.java

```diff
@@ -316,10 +198,12 @@ public Void call() throws Exception {
             // Reconnection loop
             while (running.get() && connectionFailed) {
                 try {
-                    openConnection();
+                    for (RabbitConsumer consumer : consumers) {
+                        consumer.reconnect();
```

Comment:
This used to leak threads... openConnection would leave the old consumers running and start new ones. Now it just tries to reconnect the existing consumers.